test/unit/storage/test_common_xml_collections.rb

```diff
@@ -36,4 +37,14 @@ def test_collection_get_arguments
                       "#{klass} should have at most 1 required parameter in get()")
     end
   end
+
+  def test_metadata
+    attr = { key: 'test-file', body: "hello world\x00" }
```

Comment:
This NULL byte doesn't seem to cause the mocked `put_object` to fail on `f.save`. It looks like the mocked version doesn't run:

```ruby
 headers = data[:headers].merge!(options)
``` 

To keep this pull request small, I avoided updating the mocked version to look more like the real version.